utils/dbByteArrayEncoders.go

```diff
@@ -89,8 +89,8 @@ func DecodeOctal(data []byte) ([]byte, error) {
 
 // DecodeEscaped with hex or octal encodings
 func DecodeEscaped(data []byte) ([]byte, error) {
-	hexdata := data[2:]
-	if bytes.Equal(data[:2], []byte{'\\', 'x'}) {
+	if len(data) > 2 && bytes.Equal(data[:2], []byte{'\\', 'x'}) {
```

Comment:
what is data len < 2?

Reply:
there check of escape type. hex encoding or not. second case printable chars used as is and string may contain <2 characters that are printable and will be correct

Reply:
hex encoded must have prefix \x before encoded string so here we check it

Reply:
ok, got it, thankx